### PR TITLE
Configurability

### PR DESCRIPTION
**Pull request by tsibley**
Add configurability for various constants, motivated by our [work with CDC](https://github.com/nextstrain/private/issues/94). See commits for details.

## Checklist

- [x] Checks pass

### CODE REVIEW

File: src/aws.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * AWS bits and bobs.
+ *
+ * @module aws
+ */
+import { NodeHttpHandler } from "@smithy/node-http-handler";
```
Comment (victorlin) [inline src/aws.js:6]: Suggestion: Add `@smithy/node-http-handler` to the list of direct dependencies in package.json. Currently it's an indirect dependency through `@aws-sdk/client-s3`.

**Review by victorlin — CHANGES_REQUESTED** [review]

File: env/production/outputs-extra.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * session hijacking via cookie theft.  The big mitigation for me is that this
+ * cookie is already HTTP-only, so JS injections on those providers (which are
+ * much much more likely than server-side injections) can't access the cookie.
+ * The comprehensive, longer term solution (that everyone serious about
+ * security does for essentially this same reason) is to move our internal
+ * services off the user-facing domain (e.g. support.nextstrain-team.org)
+ * and/or accomplish session portability another way (e.g. explicit SSO by
+ * next.nextstrain.org against nextstrain.org as an IdP instead of implicit SSO
+ * via session sharing).
+ *   -trs, 18 March 2022
```
[the remark this answers is not in the document]
Comment (tsibley) [reply env/production/outputs-extra.tf:18]: Ha, yeah. Saves a trip into `git blame` too and immediately adds a sense of how long the comment's been kicking around…

File: src/config.js [hunk 3]
```diff
@@ -226,14 +244,89 @@ if (typeof OIDC_IAT_BACKDATED_BY !== 'number') {
...
+ *       group-logo.png
+ *       group-overview.md
+ *       datasets/
+ *         ${datasetName}.json
+ *         ${datasetName}_${sidecarType}.json
+ *         …
+ *       narratives/
+ *         ${narrativeName}.md
+ *         …
+ *
```
[the remark this answers is not in the document]
Comment (tsibley) [reply src/config.js:328]: Good point. I'll update it with:

```diff
commit 16147069e7c5074cae047c2a3613911f95144189
Author: Thomas Sibley <tsibley@fredhutch.org>
Date:   Wed Oct 25 10:34:35 2023 -0700

    fixup! config: Configurability of the S3 bucket for Groups storage

diff --git a/src/config.js b/src/config.js
index 3d036324..af4980bf 100644
--- a/src/config.js
+++ b/src/config.js
@@ -315,7 +315,7 @@ function configPath(value) {
  * The layout of objects in the bucket will be like so:
  *
  *   ${bucketName}/
- *     ${groupName}/
+ *     ${groupNameA}/
  *       group-logo.png
  *       group-overview.md
  *       datasets/
@@ -325,6 +325,10 @@ function configPath(value) {
  *       narratives/
  *         ${narrativeName}.md
  *         …
+ *     ${groupNameB}/
+ *       …
+ *     …
+ *
  *
  * @type {string}
  * @default nextstrain-groups
```

File: src/config.js [hunk 4]
```diff
@@ -226,14 +244,89 @@ if (typeof OIDC_IAT_BACKDATED_BY !== 'number') {
...
  * Defaults to env/production/groups.json if {@link PRODUCTION} or
  * env/testing/groups.json otherwise.
  *
  * @type {string}
  */
 export const GROUPS_DATA_FILE =
      process.env.GROUPS_DATA_FILE
+  ?? configPath(configFile?.GROUPS_DATA_FILE)
   ?? path.join(__basedir, "env", (PRODUCTION ? "production" : "testing"), "groups.json");
```
[the remark this answers is not in the document]
Comment (tsibley) [reply src/config.js:301]: Ah, yes, this is better.

File: src/aws.js (hunk 1 above)
Comment (tsibley) [reply src/aws.js:6]: Whoops. I meant to but forgot.

Comment (tsibley) [reply src/aws.js:6]: Done in reworked commit 911dbf5b.

File: src/config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tsibley) [reply src/config.js]: Done in additional commit d70ff25c.

File: src/config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsibley) [reply src/config.js]: Done in reworked commit 4935e740.

File: env/production/config.json [hunk 5]
```diff
@@ -5,5 +5,7 @@
   "OAUTH2_LOGOUT_URL": "https://login.nextstrain.org/logout",
   "COGNITO_USER_POOL_ID": "us-east-1_Cg5rcTged",
   "OIDC_USERNAME_CLAIM": "cognito:username",
-  "OIDC_GROUPS_CLAIM": "cognito:groups"
+  "OIDC_GROUPS_CLAIM": "cognito:groups",
+  "SESSION_COOKIE_DOMAIN": "nextstrain.org",
+  "GROUPS_DATA_FILE": "groups.json"
```
Comment (victorlin) [inline env/production/config.json:10]: Suggestion: add these to the [infrastructure](https://github.com/nextstrain/nextstrain.org/blob/dc8500adc81aba198c474b144aaae0c9a7b3baa8/docs/infrastructure.md?plain=1#L71-L91) doc page.

**Review by victorlin — APPROVED** [review]

File: env/production/config.json (hunk 5 above)
Comment (tsibley) [reply env/production/config.json:10]: Done.

**Comment by tsibley** [comment]
I'll plan to merge this tomorrow.

**Pull request merged**